packages/canon/src/components/Link/Link.tsx

```diff
@@ -14,42 +14,59 @@
  * limitations under the License.
  */
 
-import React, { forwardRef } from 'react';
+import React, { forwardRef, memo, ComponentType } from 'react';
 import { useResponsiveValue } from '../../hooks/useResponsiveValue';
 import clsx from 'clsx';
 
-import type { LinkProps } from './types';
+import type { LinkProps, LinkRenderProps } from './types';
 
 /** @public */
-export const Link = forwardRef<HTMLAnchorElement, LinkProps>((props, ref) => {
-  const {
-    children,
-    variant = 'body',
-    weight = 'regular',
-    style,
-    className,
-    ...restProps
-  } = props;
-
-  // Get the responsive values for the variant and weight
-  const responsiveVariant = useResponsiveValue(variant);
-  const responsiveWeight = useResponsiveValue(weight);
-
-  return (
-    <a
-      ref={ref}
-      className={clsx(
+export const Link = memo(
+  forwardRef<HTMLAnchorElement, LinkProps>((props, ref) => {
+    const {
+      children,
+      variant = 'body',
+      weight = 'regular',
+      style,
+      className,
+      render,
+      to,
+      ...restProps
+    } = props;
+
+    const responsiveVariant = useResponsiveValue(variant);
+    const responsiveWeight = useResponsiveValue(weight);
+
+    const linkProps: LinkRenderProps = {
+      className: clsx(
         'canon-Link',
         responsiveVariant && `canon-Link--variant-${responsiveVariant}`,
         responsiveWeight && `canon-Link--weight-${responsiveWeight}`,
         className,
-      )}
-      style={style}
-      {...restProps}
-    >
-      {children}
-    </a>
-  );
-});
+      ),
+      style,
+      children,
+      to,
+      ...restProps,
+    };
+
+    if (render) {
+      // If render is a component type, wrap it in memo to prevent unnecessary re-renders
+      if (typeof render === 'function' && !render.length) {
+        const MemoizedComponent = memo(
```

Comment:
mmm but `MemoizedComponent` will be different on each re-render. I think `memo` shouldn't be used inside a render method. Also as `...linkProps` is spread, it might cause a re-render anyway...